taker-electron/src/main/endpoint.ts

```diff
@@ -0,0 +1,69 @@
+import { net } from "electron";
+import log from "electron-log";
+import nodenet from "node:net";
+
+const DEFAULT_PORT = 7113;
+
+export class Endpoint {
+    public url: string;
+
+    constructor(public port: number) {
+        this.url = `http://127.0.0.1:${port}`;
+    }
+
+    wait(onReady: (endpoint: string) => void) {
+        const alive = (timeout: number) => {
+            log.info(`Probing if ItchySats is alive at ${this.url}`);
+            const request = net.request(this.url);
+            request.on("response", () => onReady(this.url));
+            request.on("error", (error) => {
+                log.warn("Could not connect to ItchySats.");
+                log.error(error);
+                setTimeout(alive, timeout, timeout + timeout);
```

Comment:
🤔 I find the usage of `setTimeout` a bit confusing here. In this line you set it with three parameters, one being the double timeout. At the end of the function (line 27) you set it with four parameters. It's not obvious for me why or what this code achieves. Is there a more intuitive way to do this?

Reply:
actually the 4 parameters was one too many.. I've fixed that. The first timeout is the amount of time that should be waited until alive will be called. the second timeout represents how much the the next alive check should wait. This way we a debouncing the alive checks and prevent endless http requests. e.g. starting with 500ms the interval would look like this. 1. 500, 2. 1000, 3. 2000, 4. 4000, 5. 8000, ...

Its a recursive function, I guess that makes it a bit more confusing.